framework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -0,0 +1,202 @@
+      featData = self._getDataFromDataDict(datasets, feat, names)
+      targData = self._getDataFromDataDict(datasets, targ, names)
+      msrData = self._getDataFromDataDict(datasets, msr, names)
+
+    # Values from first entry of tuple data from _getDataFromDataDict
+    yExp = np.array(featData[0])
+    yApp = np.array(targData[0])
+    yMsr = np.array(msrData[0])
+    # Sample mean as reference value
+    yExpRef = np.mean(yExp, axis=0)
+    yAppRef = np.mean(yApp)
+    # Standardization
+    yExpStd = (yExp-yExpRef)/yExpRef
+    yAppStd = (yApp-yAppRef)/yAppRef
+    if yExp.shape[1]!= yMsr.shape[1]:
+      self.raiseAnError(IOError, "Number of Measurements is not consistent with number of Experiments/Features.")
+    yMsrStd = (yMsr-yExpRef)/yExpRef
+
+    # Single Experiment response
+    if yExpStd.shape[1]==1:
+      yExpReg = yExpStd.flatten()
+    # Pseudo response of multiple Experiment responses
+    # OrthogonalMatchingPursuit from sklearn used here
+    # Possibly change to other regressors      
+    elif yExpStd.shape[1]>1:
+      regrExp = OrthogonalMatchingPursuit(fit_intercept=False).fit(yExpStd, yAppStd)
+      yExpReg = regrExp.predict(yExpStd)
+    
+    # Kernel Desnity Estimation
+    m1 = yExpReg[:]
+    m2 = yAppStd.flatten()
+    xmin = m1.min()
+    xmax = m1.max()
+    ymin = m2.min()
+    ymax = m2.max()
+
+    binKDE = 200j
+    # Grid of Experiment (X), grid of Application (Y)
+    X, Y = np.mgrid[xmin:xmax:binKDE, ymin:ymax:binKDE]
+    psts = np.vstack([X.ravel(), Y.ravel()])
+    vals = np.vstack([m1, m2])
+    # kernel
+    knl = stats.gaussian_kde(vals)
+    # Joint PDF of Experiment and Application
+    Z = np.reshape(knl(psts).T, X.shape)
+
+    if yMsrStd.shape[1]==1:
+      yMsrStd = yMsrStd.flatten()
+    # Combine measurements by multiple Experiment regression
+    elif yMsrStd.shape[1]>1:
+      yMsrStd = regrExp.predict(yMsrStd)
+    
+    # Measurement PDF with KDE
+    knlMsr = stats.gaussian_kde(yMsrStd)
+    pdfMsr = knlMsr(X[:, 0])
+
+    # yAppPred by integrating p(yexp, yapp)dyexp * p(yexp)
+    # on range [ymsr.min(), ymsr.max()]
+    pdfAppPred = np.zeros(Y.shape[1])
+    intgrPdf = 0.0 # for normalization
+    for i in range(len(Y[0, :])):
+      pAppi = 0.0
+      for j in range(len(X[:, 0])):
+        pAppi += knl.evaluate([X[j, 0], Y[0, i]]) * np.diff(X[:, 0])[0] * pdfMsr[j]
+      pdfAppPred[i] = pAppi
+      intgrPdf += pdfAppPred[i]*(Y[0, 1]-Y[0, 0])
+
+    # normalized PDF of predicted application
+    pdfAppPredNorm = pdfAppPred/intgrPdf
+
+    # Calculate Expectation (average value) of predicted application
+    predMean = 0.0
+    for i in range(len(Y[0, :])):
+      predMean += Y[0, i]*pdfAppPredNorm[i]*(Y[0, 1]-Y[0, 0])
+
+    # Calculate Variance of predicted application
+    predVar = 0.0
+    for i in range(len(Y[0, :])):
+      predVar += (Y[0, i]-predMean)**2.0 * pdfAppPredNorm[i]*(Y[0, 1]-Y[0, 0])
```

Comment:
I'm not clear about these equations, please try to prepare a document for these calculations.

Reply:
Changed to a static model.

Reply:
Used the expressions of expected value and variance from continuous PDF.